go/mysql/sql_error.go

```diff
@@ -199,6 +199,26 @@ var stateToMysqlCode = map[vterrors.State]struct {
 	vterrors.WrongValueCountOnRow:         {num: ERWrongValueCountOnRow, state: SSWrongValueCountOnRow},
 }
 
+// ConvertStateToMySQLErrorCode returns MySQL error code for the given vterrors.State
+// If the state is == 0, an empty string is returned
+func ConvertStateToMySQLErrorCode(state vterrors.State) string {
+	if state == 0 {
+		return ""
+	}
+	s := stateToMysqlCode[state]
```

Comment:
`ConvertStateToMySQLErrorCode` and `ConvertStateToMySQLState` are slightly different. The former returns the error number as a string. While the latter returns the error state as a string. For instance, the `Undefined` error would return:
- With `ConvertStateToMySQLErrorCode` -> `"1105"`
- With `ConvertStateToMySQLState` -> `"HY000"`

Those two methods are only used by the vterrors template as internal template function. I decided to narrow down the responsibility and return value of those methods to ease the use of the functions inside the templates. Here is the snippet of how we add the methods to the template:
```go
	t.Funcs(map[string]any{
		"ConvertStateToMySQLErrorCode": mysql.ConvertStateToMySQLErrorCode,
		"ConvertStateToMySQLState":     mysql.ConvertStateToMySQLState,
		"FormatError": func(err error) string {
			s := err.Error()
			return strings.TrimSpace(strings.Join(strings.Split(s, ":")[1:], ":"))
		},
	})
```

Reply:
Ah yes you can't simply call it because it returns something.
My point is that 4 lines of code are identical between these two functions, they need to be re-used instead of duplicated.

Reply:
Will do

Reply:
Fixed in 865dffb48e